packages/babel-plugin-transform-simplify-comparison-operators/src/index.js

```diff
@@ -1,6 +1,25 @@
 "use strict";
 
-module.exports = function() {
+module.exports = function({ types: t }) {
+  function isPureVoid(path) {
+    return path.isUnaryExpression({ operator: "void" }) && path.isPure();
+  }
+
+  function isRealUndefined(path) {
+    return (
+      path.isIdentifier({ name: "undefined" }) &&
+      !path.scope.getBinding("undefined")
```

Comment:
why is this check required?. Even if someone does 
```js
var undefined = 'blah';
```

Its a read only property and does not impact our null conversion.